cli-plugins/socket/socket.go

```diff
@@ -14,21 +14,23 @@ import (
 const EnvKey = "DOCKER_CLI_PLUGIN_SOCKET"
 
 // SetupConn sets up a Unix socket listener, establishes a goroutine to handle connections
-// and update the conn pointer, and returns the environment variable to pass to the plugin.
-func SetupConn(conn **net.UnixConn) (string, error) {
+// and update the conn pointer, and returns the environment variable to pass to the plugin
+// as well as the listener's close function
+func SetupConn(conn **net.UnixConn) (string, func(), error) {
 	listener, err := listen("docker_cli_" + uuid.Generate().String())
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	accept(listener, conn)
 
-	return EnvKey + "=" + listener.Addr().String(), nil
+	closeListener := func() {
+		listener.Close()
+	}
+	return EnvKey + "=" + listener.Addr().String(), closeListener, nil
```

Comment:
Starting to wonder now if it was a good choice to change the signature to return a string (`env=address`) or if we should revert to making it return the actual listener (and have the caller decide what to do with it, and to handle the `Close()`.


No strong opinion either way; mostly thinking out loud here; what do you think feels more "natural" as a signature?

Reply:
Yeah, I debated with the same question while fixing the early defer – since we don't have to abstract the "env value determination process" away in platform-specific implementations, it might be a lot cleaner to just return the listener.